### PR TITLE
shfmt

### PR DESCRIPTION
**Pull request by kron4eg**
* format shell scripts in hack
* verify that format is maintained

**What this PR does / why we need it**:
To remove any arguing about shell style (what gofmt did to go)

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/changelog-gen.sh [hunk 1]
```diff
@@ -49,31 +51,31 @@ if [ "$(git rev-parse --abbrev-ref HEAD)" == "master" ]; then
...
 fi
 
 # Create CHANGELOG.md if not exists in order for the cat not to fail
-[ -f $CHANGELOG_FILE ] ||  touch $CHANGELOG_FILE
+[ -f $CHANGELOG_FILE ] || touch $CHANGELOG_FILE
 
 # Generate changelog
 OUTPUT="$(gchl --for-version $NEW_RELEASE since $(git describe --abbrev=0 --tags) --release-notes)"
-echo "${OUTPUT}" | cat - $CHANGELOG_FILE > temp
+echo "${OUTPUT}" | cat - $CHANGELOG_FILE >temp
```
Comment (xrstf) [inline hack/changelog-gen.sh]: Why does shfmt do The Right Thing by putting spaces around pipes, but does The Bad Thing by not having spaces for output redirects? Why is `a | b` right, but `a > b` is wrong and needs to be `a >b`?

Comment (kron4eg) [reply hack/changelog-gen.sh]: https://github.com/mvdan/sh/issues/243#issuecomment-392019995

Comment (xrstf) [reply hack/changelog-gen.sh]: Meh, I am not really convinced, but this is not a hill I want to die on. You can have your redirects, as long as we never ever indent shell scripts with tabs.

Comment (kron4eg) [reply hack/changelog-gen.sh]: There is now a flag (`-sr`) specified both prow and Makefile

**Pull request merged**